### ISSUE TITLE
Readonly vs. litteral type breaking change?

### ISSUE THREAD

**Issue by stephanedr** [Microsoft/TypeScript#12148]
**TypeScript Version:**  2.1.1

**Code**
```ts
class A {
    readonly n = 0;
    constructor() {
        this.n = 1;  // Type '1' is not assignable to type '0'
    }
}

class B {
    constructor(readonly n = 0) {}
}

class C extends B {
    constructor() {
        super(1);  // Argument of type '1' is not assignable to parameter of type '0 | undefined'.
    }
}
```

**Expected behavior:**
Is this behavior expected (breaking change vs 2.0.8)?
To me this should be valid, particularly the 2nd case.

**Actual behavior:** 
Errors displayed in comments.

**Comment by falsandtru** [comment]
#11409

**Comment by mhegazy** [comment]
All parameter initializes should be widened to their base (e.g. class `B`), property initializes (e.g. class `A`) are intentional breaking change.

**Comment by sandersn** [comment]
Fix is up at #12312
